dbtsl/api/shared/query_params.py

```diff
@@ -1,4 +1,30 @@
-from typing import List, TypedDict
+from dataclasses import dataclass
+from typing import List, Optional, TypedDict, Union
+
+from dbtsl.models.time import TimeGranularity
+
+
+@dataclass(frozen=True)
+class OrderByMetric:
+    """Spec for ordering by a metric."""
+
+    name: str
+    descending: bool = False
+
+
+@dataclass(frozen=True)
+class OrderByDimension:
+    """Spec for ordering by a dimension.
+
+    Not specifying a grain will defer the grain choice to the server.
+    """
+
+    name: str
+    grain: Optional[TimeGranularity]
```

Comment:
may need to update this (along with whereever else uses `TimeGranularity`), since I believe with custom granularity, this can be an arbitrary string.  cc: @courtneyholcomb to confirm

Reply:
that's correct!

Reply:
yep! I wanted to do that in a followup PR to avoid mixing things  but I'm aware :)